dbt/adapters/snowflake/connections.py

```diff
@@ -237,7 +237,13 @@ def exception_handler(self, sql):
         try:
             yield
         except snowflake.connector.errors.ProgrammingError as e:
-            msg = str(e)
+            unscrubbed_msg = str(e)
+
+            # A class of Snowflake errors -- such as a failure from attempting to merge
+            # duplicate rows -- includes row values in the error message, i.e.
+            # [12345, "col_a_value", "col_b_value", etc...]. We don't want to log potentially
+            # sensitive user data.
+            msg = re.sub(r"Row Values: \[.*\]", "Row Values: [redacted]", unscrubbed_msg)
```

Comment:
nit: slight performance optimization would be to use `re.compile(r"Row Values: \[.*\]", "Row Values: [redacted]")` as a global value